### PR TITLE
Register Subscribers After Node Is Synced

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Feature Addition

**What does this PR do? Why is it needed?**

 As part of #6622 , we will need to enable gossip scoring. However gossip peer scoring relies on unsynced nodes staying out
of the mesh until they are synced. This PR registers the relevant subscribers later, and only joins the mesh for the relevant topics after the node is synced.

**Which issues(s) does this PR fix?**

N.A

**Other notes for review**

### CODE REVIEW

File: beacon-chain/sync/service.go [hunk 1]
```diff
@@ -234,27 +234,41 @@ func (s *Service) registerHandlers() {
...
 				s.registerSubscribers()
 
-				if data.StartTime.After(timeutils.Now()) {
-					stateSub.Unsubscribe()
-					time.Sleep(timeutils.Until(data.StartTime))
-				}
-				log.WithField("starttime", data.StartTime).Debug("Chain started in sync service")
-				s.chainStarted = true
+				// Exit once we received the synced event.
+				stateSub.Unsubscribe()
```
Comment (farazdagi) [inline beacon-chain/sync/service.go:270]: When you return, deferred `stateSub.Unsubscribe()` will be called.

**Comment by farazdagi** [comment]
Nishant, per your request to pre-review: except for that double unsubscribe (which, I guess, is a no-op, on the second invocation) everything looks solid.

**Review by farazdagi — APPROVED** [review]

File: beacon-chain/sync/service.go [hunk 2]
```diff
@@ -234,27 +234,38 @@ func (s *Service) registerHandlers() {
...
-					stateSub.Unsubscribe()
-					time.Sleep(timeutils.Until(data.StartTime))
+				// Wait for chainstart in separate routine.
+				go func() {
+					if startTime.After(timeutils.Now()) {
+						time.Sleep(timeutils.Until(startTime))
+					}
+					log.WithField("starttime", startTime).Debug("Chain started in sync service")
+					s.chainStarted = true
+				}()
```
Comment (prestonvanloon) [inline beacon-chain/sync/service.go:259]: Do you need to return after this go routine? Otherwise it's still in the for loop, right?

**Review by prestonvanloon — CHANGES_REQUESTED** [review]

File: beacon-chain/sync/service.go (hunk 2 above)
Comment (nisdas) [reply beacon-chain/sync/service.go:259]: We need it to continue looping as we also need to receive the synced event before
exiting.

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**